awscli/customizations/cloudformation/artifact_exporter.py

```diff
@@ -47,6 +47,13 @@ def is_s3_url(url):
         return False
 
 
+def is_https_s3_url(url):
+    if isinstance(url, six.string_types) \
+        and url.startswith("https://s3.amazonaws.com/"):
```

Comment:
I didn't think so because their documentation states:

> The template must be stored on an Amazon S3 bucket, so the URL must have the form: https://s3.amazonaws.com/.../TemplateName.template

After testing it looks like CloudFormation does allow the virtual host style urls.

The two approaches I see here would be to simply test for `url.startswith('https')` or `re.match('https://[\w-]+.s3.amazonaws.com/', url)`, Thoughts?